### DISCUSSION TITLE
Shared media folder processing on all relevant stations

### DISCUSSION THREAD

**Discussion by xewl** [AzuraCast/AzuraCast#5818] [Ideas]
It's possible to share the media folder (storage location) with multiple stations.

Where I'm hitting an issue is:

When I upload a song in 1 of eg. 3 stations, sharing that folder,
 the other stations do not know about this new song, until:
 - I either manually (re-)process 1 or more files on that other station
 - or delete/edit 1 or more files on that other station
 - upload another song
 
**Note:** The sync task `CheckMediaTask` just states that new file is "unchanged", so it won't do anything with it.
 
Can it be made possible to push an event (waterfall effect) to the other stations sharing the exact same media folder, to also reprocess the media on their ends? (eg. On upload, reprocess)

I can see two points where this could be triggered:
- CheckMediaTask sync task (initial file processing, changes)
- Reprocess task on station level (after change, upload, ..?)

**Comment by BusterNeece** [comment]
@xewl They actually share StationMedia records now as well, now that StationMedia records are tied to StorageLocations and not Stations, I think what we're running into here is a cache issue. Probably should be an issue rather than a discussion eh?

**Reply by xewl** [reply]
Alright, makes sense that it'd be cache then; as I just busted it another way; and I couldn't figure out how.

I started it as an Idea discussion first; as it might've not been an issue ;D
